language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/impl/CreateFunctionCodeAction.java

```diff
@@ -74,30 +60,19 @@
         List<Object> args = Arrays.asList(lineArg, colArg, uriArg);
         Matcher matcher = CommandConstants.UNDEFINED_FUNCTION_PATTERN.matcher(diagnosticMessage);
         String functionName = (matcher.find() && matcher.groupCount() > 0) ? matcher.group(1) + "(...)" : "";
-        WorkspaceDocumentManager docManager = context.get(DocumentServiceKeys.DOC_MANAGER_KEY);
-        String diagnosedContent = getDiagnosedContent(diagnostic, context, document);
         try {
-            LSDocumentIdentifier lsDocument = docManager.getLSDocument(CommonUtil.getPathFromURI(uri).get());
-            context.put(ReferencesKeys.OFFSET_CURSOR_N_TRY_NEXT_BEST, true);
-            context.put(ReferencesKeys.DO_NOT_SKIP_NULL_SYMBOLS, true);
-            Position afterAliasPos = offsetPositionToInvocation(diagnosedContent, position);
-            // TODO: Try to use referenceAtCursor
-            SymbolReferencesModel.Reference refAtCursor = getReferenceAtCursor(context, lsDocument, afterAliasPos);
-            BLangNode bLangNode = refAtCursor.getbLangNode();
-            BLangInvocation node = null;
-            if (bLangNode instanceof BLangInvocation) {
-                node = (BLangInvocation) bLangNode;
+            Position diagPos = diagnostic.getRange().getStart();
+            NonTerminalNode cursorNode = CommonUtil.findNode(context, diagPos, CommonUtil.getPathFromURI(uri).get());
+            while (cursorNode != null &&
+                    cursorNode.kind() != SyntaxKind.MODULE_PART &&
+                    cursorNode.kind() != SyntaxKind.FUNCTION_DEFINITION &&
+                    cursorNode.kind() != SyntaxKind.FUNCTION_CALL) {
+                cursorNode = cursorNode.parent();
             }
-            if (node != null && node.pkgAlias.value.isEmpty()) {
-                boolean isWithinProject = (node.expr == null);
-                if (node.expr != null) {
-                    BLangPackage bLangPackage = context.get(DocumentServiceKeys.CURRENT_BLANG_PACKAGE_CONTEXT_KEY);
-                    List<String> currentModules = document.getProjectModules();
-                    PackageID nodePkgId = node.expr.type.tsymbol.pkgID;
-                    isWithinProject = bLangPackage.packageID.orgName.equals(nodePkgId.orgName) &&
-                            currentModules.contains(nodePkgId.name.value);
-                }
-                if (isWithinProject) {
+            if (cursorNode != null && cursorNode.kind() == SyntaxKind.FUNCTION_CALL) {
+                FunctionCallExpressionNode callExpr = (FunctionCallExpressionNode) cursorNode;
+                boolean isWithinFile = callExpr.functionName().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE;
```

Comment:
Since the name `FunctionCallExpressionNode` is bit longer; it wraps the statement into multiple lines. Thus, introduced this local variable.

`boolean isWithinFile = ((FunctionCallExpressionNode) cursorNode).functionName().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE;`